apps/server/src/app/show/show.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Model } from 'mongoose';
+import { Injectable, BadRequestException, NotFoundException } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { DocumentStatus, IShow, QueryRequest } from '@node-monorepo/types';
+
+import { ShowDocument, Show } from './show.schema';
+import { CreateShowDto } from './dto/create-show.dto';
+import { UpdateShowDto } from './dto/update-show.dto';
+import { QueryBuilder } from '../../utils/query-builder';
+import { PurchaseDto } from './dto/purchase.dto';
+import { CancelDto } from './dto/cancel.dto';
+
+@Injectable()
+export class ShowService {
+  constructor(
+    @InjectModel(Show.name)
+    private readonly ShowModel: Model<ShowDocument>
+  ) {}
+
+  create(data: CreateShowDto) {
+    return this.ShowModel.create(data);
```

Comment:
handle edge cases:
 - play not found
 - actor not found
 - available seats bigger than seats
 - Date in the past